In [2]:
import pandas as pd
import numpy as np
import seaborn as sns
import matplotlib.pyplot as plt

### Zistovanie, ci nasli sme naozaj ten povodny dataset

In [6]:
df_train = pd.read_csv('data/heart.csv',delimiter=';')
df_train['oldpeak'] = df_train['oldpeak'].str.replace(',','.')
df_train = df_train.astype(float)
df_train.rename(columns={'target,':'target'}, inplace=True)

df_test = pd.read_csv('data/heart_test.csv', delimiter=";")
df_test = df_test.astype(float)

In [8]:
df_test = df_test.loc[:, df_train.columns.tolist()]

In [9]:
df = pd.concat([df_train, df_test], axis=0).reset_index(drop=True)

In [12]:
df

age  trestbps   chol  thalach  oldpeak  target
0    63.0     145.0  233.0    150.0      2.1     1.0
1    37.0     130.0  250.0    187.0      3.5     1.0
2    41.0     130.0  204.0    172.0      1.4     1.0
3    56.0     120.0  236.0    178.0      0.8     1.0
4    57.0     120.0  354.0    163.0      0.6     1.0
..    ...       ...    ...      ...      ...     ...
298  58.0     112.0  230.0    165.0      2.5     0.0
299  50.0     150.0  243.0    128.0      2.6     0.0
300  44.0     112.0  290.0    153.0      0.0     0.0
301  60.0     130.0  253.0    144.0      1.4     0.0
302  54.0     124.0  266.0    109.0      2.2     0.0

[303 rows x 6 columns]

In [13]:
ds = pd.read_csv('heart.csv')

In [36]:
#iba stplce, ktore mame v datasete zo zadania
ds_cut = ds.loc[:, df.columns.tolist()]

In [37]:
ds_cut

age  trestbps  chol  thalach  oldpeak  target
0      52       125   212      168      1.0       0
1      53       140   203      155      3.1       0
2      70       145   174      125      2.6       0
3      61       148   203      161      0.0       0
4      62       138   294      106      1.9       0
...   ...       ...   ...      ...      ...     ...
1020   59       140   221      164      0.0       1
1021   60       125   258      141      2.8       0
1022   47       110   275      118      1.0       0
1023   50       110   254      159      0.0       1
1024   54       120   188      113      1.4       0

[1025 rows x 6 columns]

In [38]:
d = ds_cut.merge(df, how = 'outer' ,indicator=True)

In [39]:
d[d['_merge'] != "both"]

age  trestbps   chol  thalach  oldpeak  target      _merge
851   63.0     145.0  233.0    150.0      2.3     1.0   left_only
852   63.0     145.0  233.0    150.0      2.3     1.0   left_only
853   63.0     145.0  233.0    150.0      2.3     1.0   left_only
1033  63.0     145.0  233.0    150.0      2.1     1.0  right_only

### Vysledok: nesedi az jeden riadok, v ktorom hodnota oldpeak je 2.3 v povodnom datasete a 2.1 v danom nam so zadanim